Plutarch/Bool.hs

```diff
@@ -74,7 +74,7 @@ x #|| y = por # pdelay x # pdelay y
 pand :: Term s (PDelayed PBool :--> PDelayed PBool :--> PBool)
 pand = phoistAcyclic $
   plam $
-    \x y -> pif' # pforce x # (pif' # pforce y # pcon PTrue # pcon PFalse) # pcon PFalse
+    \x y -> pif (pforce x) (pif' # pforce y # pcon PTrue # pcon PFalse) (pcon PFalse)
```

Comment:
I don't think this is entirely correct either. The reason this function takes `PDelayed PBool` is that we want to avoid unnecessary delays and forces. The function is in any case quite unoptimised. What we really want if something like
```haskell
pand :: Term s (PBool :--> PDelayed PBool :--> PBool)
pand = phoistAcyclic $ plam $ \x y -> pforce $ pif' x (phoistAcyclic $ pdelay $ pcon PTrue) y
```

Reply:
Aha, of course! That's some silly redundancy from my end

Reply:
Thanks for the fix!